app/db/index.js

```diff
@@ -74,23 +87,34 @@ function addNewMarket(info) {
  * @param {*} info All question answers from the assessment page.
  */
 function updateExistingMarket(info) {
-  const marketsRef = db.child(info.marketInfo.marketName);
-
-  // Update market level.
-  marketsRef.child('marketInfo').update({
-    marketLevel: info.level,
-  });
+    // Get current values to avoid overwriting current values.
+    getSpecificMarket(info.marketInfo.marketName).then(market => {
+        const update = {
+            $set: {
+                personalInfo: {
```

Comment:
do you need to update the market info?

Reply:
You do need to update it, the $set atomic operator replaces all the info, so if a key isn't specified but exists in the database, it gets erased.  So the address and things would get deleted.

Reply:
oh ok, I thought it'd be a partial update but guess not